### ISSUE TITLE
Unable to grant spotify browse library

### ISSUE THREAD

**Issue by EPIC-Snake** [BimmerGestalt/AAIdrive#825]
Hi,

Thanks for this amazing project, I thought integration with spotify was long dead.
I'm having an issue where i'm unable to grant access to spotify library browse. Spotify search is checked and seems okay though. Last error message i received is "result was not delivered on time".
I'm using Samsung s22+ with Android 14.
Spotify version no. 8.9.78.513.
I've tried reinstall of both Spotify and AAIdrive and making new playlists and tuning settings.

This app is so close in making my dear i3 feel "modern" again, thanks again!

**Comment by hufman** [comment]
Android's RPC system is quirky. Make sure Spotify is able to run in the background to receive the app control connection. Spotify Search just goes through the web API.

**Comment by EPIC-Snake** [comment]
I've elevated all permissions and tried enabling and disabling spotify settings. Battery optimalization for both apps have been set to unlimited use and background use is allowed. When triggering the grant button to spotify library a notification pops up from spotify saying that AAIdrive is trying to connect, but it never resolves and sometimes i get the timeout error message in AAIdrive. Since authentication with the API seems to work, would it be possible to move the library browsing or playlist browsing to the API as well?

**Comment by EPIC-Snake** [comment]
Tried logging in and out of spotify again and logging in while AAIdrive notification of connecting to spotify was active and in background and suddenly i got a popup message to allow AAIdrive access to spotify.

Seems it might be fixed

**Comment by hufman** [comment]
Glad it works! 
I can't use Spotify Web to control the app while the car is driving around and might not have Internet, which is the main goal.

**Comment by TimboInOz** [comment]
Hi @hufman - I am having the same issue but with my Pixel 9 Pro Fold. Any idea how to allow the pop up from AAIdrive to Spotify? I have tried logging out and back in as per @EPIC-Snake but without success. Any ideas?

**Comment by hufman** [comment]
That popup comes directly from Spotify, and AAIdrive can do nothing more than just ask for authentication. You could check the [Spotify Authorized Apps](https://www.spotify.com/account/apps/) page to see if AAIdrive is there and you could try removing it and trying again.

**Comment by TimboInOz** [comment]
Thanks hufman - but that didn't have any impact.

**Comment by TimboInOz** [comment]
Hi hufman - I managed to resolve the issue. The pop ups were appearing but they just indicated the AAIDrive was attempting to connect. How I resolved it was to as Google to play a playlist on Spotify. As soon as I did that the permissions screen appeared and I added AAIDrive and voilà - the browse library access was granted

**Comment by hufman** [comment]
Hmmm I wonder if the reason is due to Android's restrictions on what apps are allowed to launch activities from the background, and this behavior [changed](https://developer.android.com/guide/components/activities/background-starts#:~:text=Note%3A-,Starting%20from%20Android%2014,-%2C%20if%20the%20app) for apps targetting Android 14. I see there is an [updated Spotify SDK version](https://github.com/spotify/android-sdk/releases) that changes the connection method, let me post a new version for you to test.

**Comment by michalzemko** [comment]
I managed to grant access to Spotify library using floating windows on Android, switching between Spotify and AAI, and then confirmation popup in Spotify appeared.
Now I can browse my library, select playlist. 

But browsing - search on Spotify via AAI still get empty results. So I can't search and play any other song out of (my) playlist.
Using (old) AAIdrive 1.4.3-13-b2d1395

@hufman where can I find new version to test?

**Comment by hufman** [comment]
@michalzemko @vsochna-commits @TimboInOz Please test [the latest build](https://bimmergestalt.s3.amazonaws.com/aaidrive/builds/androidautoidrive-latest-main-mapbox-sentry-release.apk) which has the updated Spotify SDK. You'll need to uninstall the Google App Store version first.

**Comment by vsochna-commits** [comment]
@michalzemko : Ability to search Spotify - this opened a new WWW window where I enabled AAIdrive. The permission is then visible via WWW Spotify in Applications

**Comment by michalzemko** [comment]
@vsochna-commits I tried to authorize "Ability to search Spotify" in AAIdrive -> opened Spotify web (logged in), clicked Agrree to authorize -> web window closed, back to AAI drive, "Ability to search Spotify" checkbox is still not checked -> on AAI opened spotify (plugin), tried to search gets <No Results Found>

Opened Spotify web in browser, I see permissions for AAI drive are enabled (Control over Spotify in BMW/Mini infotainment)

**Comment by vsochna-commits** [comment]
@michalzemko : Then I would delete the Spotify + AAIdrive cache, or just another version of AAIdrive

**Comment by michalzemko** [comment]
Cache clear helped. Thanks @vsochna-commits

**Comment by hufman** [comment]
Note that the "Ability to search Spotify" uses the Spotify Web API, which is separate from the Spotify SDK to control the local app, and the permissions are managed separately.

**Comment by vsochna-commits** [comment]
@hufman : New version:

I uninstalled 1.3.2.85-4891cd2. I installed the latest version from the link (1.4.3-38-934ed8b).

The ability to browse the Spotify library - was already active immediately after installation (so I didn't have to turn it on).

I clicked Enabled for the Spotify search option.

Everything works on a Samsung S25 and the latest version of Spotify Premium from Google Play. I'll try it in the car tonight, but it should work.

Thank you very much for the lightning fast response and willingness!!!
